lua/neoclip/telescope.lua

```diff
@@ -14,33 +14,42 @@ local settings = require('neoclip.settings').get()
 local utils = require('neoclip.utils')
 local picker_utils = require('neoclip.picker_utils')
 
-local function get_set_register_handler(register_names)
+local function refresh_entry_if_needed (typ, entry)
+    if settings.refresh_entry_on_select then
+        storage.set_as_most_recent(typ, entry)
+    end
+end
+
+local function get_set_register_handler(register_names, typ)
     return function(prompt_bufnr)
         local entry = action_state.get_selected_entry()
+        refresh_entry_if_needed(typ, entry.db_entry)
```

Comment:
[nit] I guess not important but in the fzf implementation this happened after setting the register

Reply:
That's fair. I'll make them consistent

Reply:
I'll move the `refresh_entry_if_needed` function to the bottom of all handlers. I think the entry should only be refreshed if the user action was actually successful. If some error occurred along the way, it should stay as it is.